go-controller/pkg/ovn/ovn.go

```diff
@@ -738,6 +793,127 @@ func (oc *Controller) WatchEgressFirewall() *factory.Handler {
 	}, oc.syncEgressFirewall)
 }
 
+// WatchDNSObject start the watching of dnsObject resource and calls back the
+// appropriate handler logic
+// the addressSets that we are updating is based on the dnsName so that can be generated on the fly
+func (oc *Controller) WatchDNSObject() *factory.Handler {
+	return oc.watchFactory.AddDNSObjectHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
```

Comment:
This is hard to follow. Can you make egressfirewallDNSInfo a type and append methods to it?
I think readability will benefit

Reply:
done